src/middleware/ErrorHandlingMiddleware.ts

```diff
@@ -0,0 +1,8 @@
+import ApiErrors from "../errors/ApiError.ts";
+
+export function errorHandler(err, req, res, next) {
+  if (err instanceof ApiErrors) {
+    return res.status(err.status).json({ message: err.message });
+  }
+  return res.status(500).json({ message: "Unexpected error" });
```

Comment:
you don't provide any information about this 500  error - it could complicate troubleshouting